src/coreclr/vm/nativelibrary.cpp

```diff
@@ -490,6 +498,15 @@ namespace
             hmod = LocalLoadLibraryHelper(path, flags, pErrorTracker);
         }
 
+        // Bundle with additional files extracted - also treat the extraction path as the assembly directory for native library load
+        if (hmod == NULL && isBundledAssembly && !Bundle::AppBundle->ExtractionPath().IsEmpty())
+        {
+            path.Set(Bundle::AppBundle->ExtractionPath());
+            path.Append(DIRECTORY_SEPARATOR_CHAR_W);
```

Comment:
Is this defensive or do we know we need it? I'm curious if there would be utility for an `AppendIfNotPresent()` API on `SString`.

Reply:
For singlefilehost, we need it. Technically, some custom host could implement returning this property (would not expect that to be a regular use case).

Reply:
I do see three places in the runtime where we basically do an append `DIRECTORY_SEPARATOR_CHAR_W` if it doesn't already end with it.